pkg/controllers/managementuser/rbac/namespace_handler.go

```diff
@@ -88,7 +88,14 @@ func (n *nsLifecycle) Remove(obj *v1.Namespace) (runtime.Object, error) {
 
 func (n *nsLifecycle) syncNS(obj *v1.Namespace) (bool, error) {
 	// add fleet namespace to system project
-	if isFleetNamespace(obj) {
+	if isFleetNamespace(obj) &&
+		// If this is the local cluster, then only move the namespace to ths system project if the projectIDAnnotation is
+		// empty or beings with "local" (i.e. not c-). If the projectIDAnnotation begins with something other than "local"
+		// then it is likely that local cluster is the tenant cluster in a hosted Rancher setup and the namespace belongs to
+		// the system project for the cluster in the host cluster. Moving it here would only cause the namespace to be
+		// continually moved between projects forever.
+		(n.m.clusterName != "local" || obj.Annotations[projectIDAnnotation] == "" || strings.HasPrefix(obj.Annotations[projectIDAnnotation], "local")) {
```

Comment:
Is this because we have two controllers running for the same namespace, one is at local cluster and one is at downstream rancher so it will move project back and forth?

Reply:
And in this case, we always make sure the namespace is assigned to the system project in downstream rancher?

Reply:
Yes, that is correct.